src/components/MapMaplibreGl.tsx

```diff
@@ -134,7 +137,23 @@ class MapMaplibreGlInternal extends React.Component<MapMaplibreGlInternalProps,
         this.state.inspect!.render();
       }, 500);
     }
-    
+
+    if (this.props.localPMTiles) {
+      const file = this.props.localPMTiles;
+      this.state.pmtilesProtocol!.add(file); // this is necessary for non-HTTP sources
+
+      if (map) {
+        file.getMetadata().then((metadata: any) => {
```

Comment:
Does metadata has a better type? `topojson` maybe?

Reply:
Added an interface for Metadata (borrowed from pmtiles code)